umbral-pre/src/key_frag.rs

```diff
@@ -249,63 +233,67 @@ impl KeyFrag {
         let precursor = self.precursor;
 
         // We check that the commitment is well-formed
-        let correct_commitment = commitment == &u * &key;
-
-        let mut digest = SignatureDigest::new()
-            .chain_bytes(&kfrag_id)
-            .chain_point(&commitment)
-            .chain_point(&precursor)
-            .chain_bool(self.proof.delegating_key_signed)
-            .chain_bool(self.proof.receiving_key_signed);
-        if self.proof.delegating_key_signed {
-            // `delegating_pk` is guaranteed to be Some here.
-            digest = digest.chain_pubkey(&delegating_pk.unwrap());
+        if commitment != &u * &key {
+            return false;
+        }
+
+        // A shortcut, perhaps not necessary
```

Comment:
For me personally, it was easier to understand what is being passed to `hash_to_cfrag_signature` without those early returns - `none_unless` describes the relationship between `maybe_*_pk` and `self.proof.*_key_signed` very clearly.

Reply:
In #47 these returns will return specific errors (and raise exceptions in Python's case), to help user pinpoint the cause of verification failure.